src/opm/output/eclipse/ActiveIndexByColumns.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (c) 2021 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/output/eclipse/ActiveIndexByColumns.hpp>
+
+#include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+
+#include <array>
+#include <cstddef>
+#include <functional>
+#include <numeric>
+#include <vector>
+
+namespace {
+    std::size_t columnarGlobalIdx(const std::array<int, 3>& dims,
+                                  const std::array<int, 3>& ijk)
+    {
+        // Linear index assuming C-like loop order
+        //
+        //     for (i = 0 .. Nx - 1)
+        //         for (j = 0 .. Ny - 1)
+        //             for (k = 0 .. Nz - 1)
+        //
+        // as opposed to the usual Fortran-like loop order ("natural ordering")
+        //
+        //     for (k = 0 .. Nz - 1)
+        //         for (j = 0 .. Ny - 1)
+        //             for (i = 0 .. Nx - 1)
+        //
+        return ijk[2] + dims[2]*(ijk[1] + dims[1]*ijk[0]);
+    }
+}
+
+bool Opm::ActiveIndexByColumns::operator==(const ActiveIndexByColumns& rhs) const
+{
+    return this->natural2columnar_ == rhs.natural2columnar_;
+}
+
+void
+Opm::ActiveIndexByColumns::
+buildMappingTables(const std::size_t                                           numActive,
+                   const std::array<int, 3>&                                   cartDims,
+                   const std::function<std::array<int, 3>(const std::size_t)>& getIJK)
+{
```

Comment:
Is it necessary with a build method? I would guess everything could be done in the constructor?

Reply:
> I would guess everything could be done in the constructor

It's definitely **possible** to do this work in a constructor, but I'm in the process of exploring the design possibilities here and I don't yet quite know the precise details of how I'm going to use this facility (e.g., as a data member or as a construct/use/throw away object) so I'm going for the most general solution right now&mdash;create PRs early &c.  If it turns out that I can reasonably use object construction for this then I'll amend the design accordingly.